packages/react/menu/src/Menu.tsx

```diff
@@ -554,6 +584,15 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
                       searchRef.current = '';
                     }
                   })}
+                  onMouseMove={composeEventHandlers(props.onMouseMove, (event) => {
+                    if (event.currentTarget.contains(event.target as HTMLElement)) {
+                      if (lastPointerXRef.current !== event.clientX) {
+                        pointerXDirRef.current =
+                          event.clientX > lastPointerXRef.current ? 'right' : 'left';
+                        lastPointerXRef.current = event.clientX;
+                      }
+                    }
```

Comment:
`onMouseMove` fires so often that it'll report the same value multiple times, this was a simple way to only update the direction if they've actually changed. The outer most condition is to just avoid it bubbling and updating needlessly up the tree.

Reply:
This is all actually making me think that rather than tracking the pointer direction, and use the mouse leave event coordinates, if we tracked the pointer position in a ref in mousemove, we probably wouldn't need the artificial bleed and we wouldn't need the directional check, I have a feeling it would all work actually better this way and the code would be simpler.

Reply:
Me and @benoitgrelard went into more detail with this on a call, it was something I had tried before but it had some non obvious caveats that meant we'd still need to rely on pointer movement direction. We decided to stick to the original implementation but tidy it up.